### PR TITLE
Improved docs for range partitioning.

### PR DESCRIPTION
**Pull request by gianm**
1) Clarify the benefits of range partitioning.
2) Clarify which filters support pruning.
3) Include the fact that multi-value dimensions cannot be used for partitioning.

### CODE REVIEW

File: docs/ingestion/native-batch.md [hunk 1]
```diff
@@ -345,10 +345,22 @@ The Parallel task will use one subtask when you set `maxNumConcurrentSubTasks` t
...
+2. Better query performance due to Broker-level segment pruning, which removes segments from
+   consideration when they cannot possibly contain data matching the query filter.
+
+For Broker-level segment pruning to be effective, you must include the `partitionDimension` in the
+`WHERE` clause at query time, using a filter that supports pruning. Filters that support pruning
+include:
+
+- Equality, like `x = 'foo'` or `x IN ('foo', 'bar')`
+- Comparison on strings, like `x < 'foo'` or other comparisons involving `<`, `>`, `<=`, or `>=`.
+  Comparisons on numbers do not support pruning.
```
Comment (suneet-s) [inline docs/ingestion/native-batch.md:360]: Does this mean Equality on numbers like `x = 123` supports segment pruning?

Comment (gianm) [reply docs/ingestion/native-batch.md]: Good question. It depends on the type of `x`. I updated the language to be clearer.

**Comment by techdocsmith** [comment]
@gianm, thanks for getting this into the doc. Is there any way we could organize this such that we don't repeat the exact same benefits for both single dim and multi dim partitioning?

I realize we're already deep into headers at this point, but would it make sense to have a "Range partitioning" header with the benefits, then sub-divide for singled-dim then multi-dim?

**Comment by kfaraz** [comment]
@techdocsmith , @gianm , do we even need separate sections for single-dim and multi-dim anymore?
We could just have section for "Range partitioning" and add a note that single-dim is nothing but range with only one dimension.

File: docs/ingestion/native-batch.md [hunk 2]
```diff
@@ -387,19 +400,44 @@ them to create the final segments. Finally, they push the final segments to the
...
+
+1. Lower storage footprint due to combining similar data into the same segments, which improves compressibility.
+2. Better query performance due to Broker-level segment pruning, which removes segments from
+   consideration when they cannot possibly contain data matching the query filter.
+
+For Broker-level segment pruning to be effective, you must include a set of `partitionDimensions`,
+starting from the left, in the `WHERE` clause at query time using filters that support pruning.
+Filters that support pruning include:
+
+- Equality on literals that match the type of the column, like `x = 'foo'` and `x IN ('foo', 'bar')`
```
Comment (kfaraz) [inline docs/ingestion/native-batch.md:421]: Isn't type of the dimensions always string (atleast for now)?
Also, the pruning happens assuming a RangeSet of String using string comparisons only, even for equality.

Please correct me if I am missing something.

File: docs/ingestion/native-batch.md [hunk 3]
```diff
@@ -387,19 +400,44 @@ them to create the final segments. Finally, they push the final segments to the
...
+  where `x` is a string; or `x = 3` and `x IN (2, 3)` where `x` is numeric.
+- Comparison between string columns and string literals, like `x < 'foo'` or other comparisons
+  involving `<`, `>`, `<=`, or `>=`. Comparisons involving numbers do not support pruning.
```
Comment (kfaraz) [inline docs/ingestion/native-batch.md:424]: This part is probably not needed, if the above comment regarding type of dimensions always being string holds true.

```suggestion
  involving `<`, `>`, `<=`, or `>=`.
```

**Comment by techdocsmith** [comment]
@kfaraz +1 for consolidating

File: docs/ingestion/native-batch.md [hunk 4]
```diff
@@ -387,19 +400,44 @@ them to create the final segments. Finally, they push the final segments to the
...
-For segment pruning to be effective and translate into better query performance, you must include the first of your `partitionDimensions` in the `WHERE` clause at query time. For example, given the following `partitionDimensions`:
-```
- "partitionsSpec": {
-        "type": "range",
-        "partitionDimensions":["coutryName","cityName"],
-        "targetRowsPerSegment" : 5000
+When you use multi-dimension range partitioning for your data, Druid is able to distribute segment
+sizes more evenly than with single dimension partitioning.
+
+Range partitioning has several benefits:
```
Comment (ektravel) [inline docs/ingestion/native-batch.md:411]: Consider combining lines 348-364 and 411-440 and moving the text into a separate section titled **Benefits of range partitioning**.

File: docs/ingestion/native-batch.md (hunk 2 above)
Comment (gianm) [reply docs/ingestion/native-batch.md:421]: I thought that non-string types would be converted into strings and still would be included in the shard specs. IIRC you implemented it, though, so I'll believe what you say. What does happen in this case?

File: docs/ingestion/native-batch.md (hunk 4 above)
Comment (gianm) [reply docs/ingestion/native-batch.md:411]: > Consider combining lines 348-364 and 411-440 and moving the text into a separate section titled **Benefits of range partitioning**.

I made this change.

> Isn't type of the dimensions always string (atleast for now)?
> Also, the pruning happens assuming a RangeSet of String using string comparisons only, even for equality.

I made this change too. @kfaraz please let me know if the new text is correct.

**Comment by gianm** [comment]
> Isn't type of the dimensions always string (atleast for now)?
> Also, the pruning happens assuming a RangeSet of String using string comparisons only, even for equality.

I made this change too. @kfaraz please let me know if the new text is correct.

**Review by kfaraz — APPROVED** [review]
Thanks for the changes, @gianm .
LGTM 🚀

**Comment by gianm** [comment]
🎉

**Pull request merged**